src/models/document/document.ts

```diff
@@ -174,6 +175,8 @@ export const DocumentModel = types
       }
       else {
         self.content = DocumentContentModel.create(snapshot);
+        const sharedModelManager = getEnv(self).sharedModelManager as ISharedModelDocumentManager;
+        sharedModelManager.setDocument(self.content);
```

Comment:
The main reason was that the DocumentModel is the root of the MST tree. So that is where the environment needs to be added. I found that if I provided the environment when creating the DocumentContentModel and then added that to DocumentModel, MST would complain that the environments didn't match. So it was easier to just add the environment to the DocumentModel, and that way when the DocumentContentModel instance is added to the DocumentModel it could start working with the manager.

Reply:
@kswenson does this make sense now?